lib/galaxy/config/__init__.py

```diff
@@ -339,40 +333,37 @@ def _process_config(self, kwargs):
             for ip in kwargs.get("fetch_url_whitelist", "").split(',')
             if len(ip.strip()) > 0
         ]
-        self.template_path = resolve_path(kwargs.get("template_path", "templates"), self.root)
-        self.template_cache = resolve_path(kwargs.get("template_cache_path", "compiled_templates"), self.data_dir)
+        self.template_path = os.path.join(self.root, kwargs.get("template_path", "templates"))
+        self.template_cache = os.path.join(self.data_dir, self.template_cache_path)
         self.job_queue_cleanup_interval = int(kwargs.get("job_queue_cleanup_interval", "5"))
-        self.cluster_files_directory = self.resolve_path(kwargs.get("cluster_files_directory", os.path.join(self.data_dir, "pbs")))
+        self.cluster_files_directory = self.resolve_path(os.path.join(self.data_dir, self.cluster_files_directory))
 
         # Fall back to legacy job_working_directory config variable if set.
-        default_jobs_directory = kwargs.get("job_working_directory", "jobs_directory")
-        self.jobs_directory = resolve_path(kwargs.get("jobs_directory", default_jobs_directory), self.data_dir)
+        self.jobs_directory = os.path.join(self.data_dir, kwargs.get("jobs_directory", self.job_working_directory))
         if self.preserve_python_environment not in ["legacy_only", "legacy_and_local", "always"]:
             log.warning("preserve_python_environment set to unknown value [%s], defaulting to legacy_only")
             self.preserve_python_environment = "legacy_only"
         self.nodejs_path = kwargs.get("nodejs_path", None)
         # Older default container cache path, I don't think anyone is using it anymore and it wasn't documented - we
         # should probably drop the backward compatiblity to save the path check.
-        self.container_image_cache_path = resolve_path(kwargs.get("container_image_cache_path", "container_images"), self.data_dir)
+        self.container_image_cache_path = os.path.join(self.data_dir, kwargs.get("container_image_cache_path", "container_images"))
         if not os.path.exists(self.container_image_cache_path):
             self.container_image_cache_path = self.resolve_path(kwargs.get("container_image_cache_path", os.path.join(self.data_dir, "container_cache")))
         self.output_size_limit = int(kwargs.get('output_size_limit', 0))
         # activation_email was used until release_15.03
         activation_email = kwargs.get('activation_email', None)
         self.email_from = kwargs.get('email_from', activation_email)
         self.myexperiment_target_url = kwargs.get('my_experiment_target_url', 'www.myexperiment.org')
-        self.instance_resource_url = kwargs.get('instance_resource_url', None)
+
         #  Get the disposable email domains blacklist file and its contents
-        self.blacklist_location = kwargs.get('blacklist_file', None)
```

Comment:
The `config/disposable_email_blacklist.conf.sample` sample file is not usable as it is, so the default for `blacklist_file` is actually `None`.
With these changes, `ERROR:galaxy.config:CONFIGURATION ERROR: Can't open supplied blacklist file from path: /home/circleci/repo/config/disposable_email_blacklist.conf` will always be logged at startup, see e.g. https://circleci.com/gh/galaxyproject/galaxy/38709

I see 2 possible solutions:
- change the default in `lib/galaxy/webapps/galaxy/config_schema.yml` to null
- comment out the present content of `config/disposable_email_blacklist.conf.sample` and update the code below to skip lines starting with `#`

Reply:
I'll remove the default from config_schema.yaml - that will resolve to None automatically. I think it's simple and clean. 
Thanks!